pkg/rekor/identity_test.go

```diff
@@ -716,3 +849,120 @@ func TestOIDMatchesValue(t *testing.T) {
 		t.Errorf("Expected string to equal 'test cert value', got %s", extValue)
 	}
 }
+
+// Test mergeOIDMatchers
+func TestMergeOIDMatchers(t *testing.T) {
+	oidMatchers, err := mergeOIDMatchers(MonitoredValues{OIDMatchers: []OIDMatcher{{
+		ObjectIdentifier: asn1.ObjectIdentifier{2, 5, 29, 17},
+		ExtensionValues:  []string{},
+	}}})
+	if err != nil {
+		t.Errorf("Expected nil, got %v", err)
+	}
+	if len(oidMatchers) != 1 {
+		t.Errorf("Expected 1 OIDMatcher, got %d", len(oidMatchers))
+	}
+
+	oidMatchers, err = mergeOIDMatchers(MonitoredValues{OIDMatchers: []OIDMatcher{{
+		ObjectIdentifier: asn1.ObjectIdentifier{2, 5, 29, 17},
+		ExtensionValues:  []string{""},
+	}}})
+	if err != nil {
+		t.Errorf("Expected nil, got %v", err)
+	}
+	if len(oidMatchers) != 1 {
+		t.Errorf("Expected 1 OIDMatcher, got %d", len(oidMatchers))
+	}
+
+	oidMatchers, err = mergeOIDMatchers(MonitoredValues{OIDMatchers: []OIDMatcher{{
+		ObjectIdentifier: asn1.ObjectIdentifier{2, 5, 29, 17},
+		ExtensionValues:  []string{"test", "test2"},
+	}}})
+	if err != nil {
+		t.Errorf("Expected nil, got %v", err)
+	}
+	if len(oidMatchers) != 1 {
+		t.Errorf("Expected 1 OIDMatcher, got %d", len(oidMatchers))
+	}
+
+	oidMatchers, err = mergeOIDMatchers(MonitoredValues{OIDMatchers: []OIDMatcher{{
+		ObjectIdentifier: asn1.ObjectIdentifier{2, 5, 29, 17},
+		ExtensionValues:  []string{"test1"},
+	}, {
+		ObjectIdentifier: asn1.ObjectIdentifier{2, 5, 29, 17},
+		ExtensionValues:  []string{"test"},
+	}}})
+	if err != nil {
+		t.Errorf("Expected nil, got %v", err)
+	}
+	if len(oidMatchers) != 1 {
+		t.Errorf("Expected 1 OIDMatcher, got %d", len(oidMatchers))
+	}
+
+	oidMatchers, err = mergeOIDMatchers(MonitoredValues{OIDMatchers: []OIDMatcher{{
+		ObjectIdentifier: asn1.ObjectIdentifier{2, 5, 29, 17},
+		ExtensionValues:  []string{"test1"},
+	}, {
+		ObjectIdentifier: asn1.ObjectIdentifier{2, 5, 29, 18},
+		ExtensionValues:  []string{"test"},
+	}}})
+	if err != nil {
+		t.Errorf("Expected nil, got %v", err)
+	}
+	if len(oidMatchers) != 2 {
+		t.Errorf("Expected 1 OIDMatcher, got %d", len(oidMatchers))
+	}
+}
+
+// test parseObjectIdentifier
+func TestParseObjectIdentifier(t *testing.T) {
```

Comment:
This looks like a test we can write using the table tests pattern. https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
i was going to save this refactoring for a separate pr after this so i can tackle a bunch of the tests at once and convert them to table tests- i can start with this one now inside of this pr if that would be preferable

Reply:
Oh, makes sense, I was not sure if you wanted to do refactoring for old tests only or for all tests at once.